core/sdk-core/src/main/java/software/amazon/awssdk/core/RequestOverrideConfiguration.java

```diff
@@ -371,6 +388,22 @@ default B putRawQueryParameter(String name, String value) {
 
         List<MetricPublisher> metricPublishers();
 
+        /**
+         * Sets the additional execution attributes collection for this request.
+         * @param executionAttributes Execution attributes map for this request
+         * @return This object for method chaining.
+         */
+        B executionAttributes(Map<ExecutionAttribute<?>, ?> executionAttributes);
```

Comment:
Any reason we don't just accept `ExecutionAttributes`? Same for the client config